File: app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.content.res.Configuration;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.myapplication.fragments.ContentFragment;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;

public class MainActivity extends AppCompatActivity implements ContentFragment.OpenFileInputCallback {
    private final static String FILE_NAME = "content.txt";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            ContentFragment contentFragment = new ContentFragment();
            contentFragment.setOpenFileInputCallback(this);
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.fragment_container, contentFragment)
                    .commit();
        }
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig); // Remove the default behavior of recreating the fragments
    }

    public void saveText(View view) {
        
        String existedText = openText();
        StringBuilder sb = new StringBuilder(existedText);
        
        FileOutputStream fos = null;
        try {
            EditText textBox = (EditText) findViewById(R.id.textInputEditText);
            String text = textBox.getText().toString();
            sb.append("\n").append(text);
            text = sb.toString();

            fos = openFileOutput(FILE_NAME, MODE_PRIVATE);
            fos.write(text.getBytes());
            Toast.makeText(this, "Файл збережено", Toast.LENGTH_SHORT).show();
        } catch (IOException ex) {
            Toast.makeText(this, ex.getMessage(), Toast.LENGTH_SHORT).show();
        } finally {
            try {
                if (fos != null)
                    fos.close();
            } catch (IOException ex) {
                Toast.makeText(this, ex.getMessage(), Toast.LENGTH_SHORT).show();
            }
        }
    }

    @Override
    public FileInputStream openFileInputStream(String fileName) throws FileNotFoundException {
        return openFileInput(fileName);
    }

    public String openText() {
        FileInputStream fin = null;
        try {
            fin = openFileInput(FILE_NAME);
            byte[] bytes = new byte[fin.available()];
            fin.read(bytes);
            String text = new String(bytes);
            return text;
        } catch (IOException ex) {
            Toast.makeText(this, ex.getMessage(), Toast.LENGTH_SHORT).show();
        } finally {
            try {
                if (fin != null)
                    fin.close();
            } catch (IOException ex) {
                Toast.makeText(this, ex.getMessage(), Toast.LENGTH_SHORT).show();
            }
        }
        return "Error occurred.";
    }
}

File: app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.RadioButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment


class MainActivity : AppCompatActivity(), View.OnClickListener {

    fun getResponseFromCheckBox(answerIsYes: Boolean, answerIsNo: Boolean): String {
        if (!answerIsYes && !answerIsNo) {
            return "No Answer"
        }
        if (answerIsYes) {
            return "Yes"
        }
        return "No"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button = findViewById<Button>(R.id.button)
        val input = findViewById<View>(R.id.textInputEditText) as EditText
        val radioYes = findViewById<View>(R.id.radio_yes) as RadioButton
        val radioNo = findViewById<View>(R.id.radio_no) as RadioButton
        var output = findViewById<View>(R.id.output) as TextView

        button.setOnClickListener {
            val message = String.format("Question: %s \nAnswer: %s", input.text.toString(), getResponseFromCheckBox(radioYes.isChecked, radioNo.isChecked))
            Log.v("EditText", message)
            output.text = message
        }

    }

    override fun onClick(v: View?) {
        TODO("Not yet implemented")
    }
}